docs: Add a new chapter for third-party component upgrades

alfrunes opened the pull request:
Changelog: Title
Ticket: MC-6324

merlin-northern commented on 07.Server-installation/90.Component-upgrades/docs.md:
```diff
@@ -0,0 +1,39 @@
+versions. Please follow the [official upgrade
+guide](https://docs.nats.io/running-a-nats-service/nats_admin/upgrading_cluster)
+when upgrading your NATS cluster. If deploying with helm, we recommend scaling
+up the number of nodes in the cluster before upgrading the Docker image so that
```
I would re-phrase somehow the "upgrading the Docker image" part. maybe just say that we are setting a new tag, or setting a new image reference. Up to you.

merlin-northern commented on 07.Server-installation/90.Component-upgrades/docs.md:
```diff
@@ -0,0 +1,39 @@
+Mender only uses Redis as an in-memory cache service, and requires Redis > 6.0.
+Please make sure to follow Redis [release
+cycle](https://redis.io/docs/about/releases/) to stay on top of latest security
+fixes. When upgrading Redis, please note that the number of requests to MongoDB
```
there is a small repetition here, we can I think have
```suggestion
fixes. When upgrading, please note that the number of requests to MongoDB
```

merlin-northern reviewed (approved):
left some comments, als as a review checked all the links.

merlin-northern commented on 07.Server-installation/90.Component-upgrades/docs.md:
````diff
@@ -0,0 +1,39 @@
+> need to manually set the feature compatibility flag once the cluster is
+> running the new image:
+> ```js
+> // /usr/bin/mongosh:
````
instead of this comment I would just excplicitly note that this is `mongosh` either with words before this code block or with separate `# `  shell command example.

alfrunes commented on 07.Server-installation/90.Component-upgrades/docs.md:
````diff
@@ -0,0 +1,39 @@
+
+## MongoDB
+
+Mender supports both MongoDB 5.0 and 4.4. To upgrade from 4.4 to 5.0, please
+follow the upgrade procedure from the [official
+documentation](https://www.mongodb.com/docs/manual/release-notes/5.0-upgrade-replica-set).
+
+> NOTE: When deploying the MongoDB using helm chart, this upgrade can be 
+> performed transparently by upgrading the Docker image version. However, you
+> need to manually set the feature compatibility flag once the cluster is
+> running the new image:
+> ```js
+> // /usr/bin/mongosh:
+> db.adminCommand({setFeatureCompatibilityVersion: "5.0"})
+> ```
````
A maintenance window is not required if you're following the docs and upgrading the secondaries first and then stepping down the primary. As long as the primary and one secondary is running, the cluster is available.
But for helm I should probably also mention that you should set an appropriate PodDisruptionBudget and UpdateStrategy when performing the upgrade.

merlin-northern replied:
> A maintenance window is not required if you're following the docs and upgrading the secondaries first and then stepping down the primary. As long as the primary and one secondary is running, the cluster is available. 

as you can tell looking at this lousy reviewer comments it maybe nice to mention that you _dont_ need the maintenance window. (pretty much what you said maybe nice to have in the docs)

> But for helm I should probably also mention that you should set an appropriate PodDisruptionBudget and UpdateStrategy when performing the upgrade.

works for me

TheMeaningfulEngineer reviewed (changes requested):
Please define supported version only in [this section](https://github.com/mendersoftware/mender-docs/blob/master/07.Server-installation/04.Production-installation-with-kubernetes/docs.md)

The reason for this is to provide a simple reference on what is considered supported for  on-prem.

Please don't use phrasing which mixes 'supported' and 'should work':

> Although Mender only requires NATS >= 2.2, we recommend keeping NATS as close to
the latest release

What we run on Hosted Mender and in the testing pipelines is supported this should be clearly separated from things that should 'in theory' work.

alfrunes commented:
@merlin-northern I improved the documentation for MongoDB, adding a new section covering the helm chart.

@TheMeaningfulEngineer I updated the supported versions section, and adding a pointer to the new upgrade chapter.

alfrunes commented:
@TheMeaningfulEngineer I updated the docs as specified. I also did a reality check with the upgrade documentations.

TheMeaningfulEngineer commented on 07.Server-installation/04.Production-installation-with-kubernetes/docs.md:
```diff
@@ -37,17 +37,27 @@ The requirements listed below are what Northern.tech tests internally and suppor
+    * MongoDB 5.x (Atlas rapid releases)
+  * MongoDB 4.4
+
+!!! To upgrade MongoDB, please refer to the [official
+!!! documentation](https://www.mongodb.com/docs/manual/release-notes/5.0-upgrade-replica-set/).
```
```suggestion
```

It's already mentioned in the `Upgrading to 3.6` section

TheMeaningfulEngineer commented on 07.Server-installation/04.Production-installation-with-kubernetes/docs.md:
```diff
@@ -37,17 +37,27 @@ The requirements listed below are what Northern.tech tests internally and suppor
+service.
+* Supported versions:
+  * MongoDB 5.0
+    * MongoDB 5.x (Atlas rapid releases)
+  * MongoDB 4.4
+
+!!! To upgrade MongoDB, please refer to the [official
+!!! documentation](https://www.mongodb.com/docs/manual/release-notes/5.0-upgrade-replica-set/).
 
 **Redis:**
-* Redis 6.0
+* Redis 6.0, 6.2
   * ElasticCache (AWS)
   * AzureCache
   * Momory store (GCP)
   * Self hosted
 
+**Nats:**
+* NATS 2.7, 2.8, 2.9
```
What do you think about defining this as a minimal major version?
So instead of listing them one by one, how about:

> Redis 6.0 and above (assuming no change in the Major version)
> NATS 2.7 and above (assuming no change in the Major version)

alfrunes replied:
I think I'll just list the known compatible version (as we do for the other services) for now.

TheMeaningfulEngineer commented on 07.Server-installation/04.Production-installation-with-kubernetes/docs.md:
```diff
@@ -37,17 +37,27 @@ The requirements listed below are what Northern.tech tests internally and suppor
-* MongoDB 4.4
-  * Self hosted, both single node and replica set (recommended)
-  * Atlas
+Mender supports MongoDB in both Standalone (single node) and Replica set mode.
+We recommend using a replica set for high availability production environments,
+and you can choose to host your own cluster or use the MongoDB Atlas managed
+service.
```
```suggestion

Mender supports MongoDB in both Standalone (single node) and Replica set mode.
We recommend using a replica set for high availability production environments,
and you can choose to host your own cluster or use the MongoDB Atlas managed
service.
```

Just a line to maintain consistency with other rendered topics

TheMeaningfulEngineer replied:
```suggestion

Mender supports MongoDB in both Standalone (single node) and Replica set mode.
We recommend using a replica set for high availability production environments,
and you can choose to host your own cluster or use the MongoDB Atlas managed
service.
```

Just a line to maintain consistency with other rendered topics

alfrunes commented:
@TheMeaningfulEngineer Thanks for the feedback. I made some [minor changes](https://github.com/mendersoftware/mender-docs/compare/2480628442ade59912d243d1206ea533cdf64187..ed623df4f635ea11b4007b1d81df509daba348f5) addressing your comments. 
Could you please have a look?

TheMeaningfulEngineer reviewed (approved):
Looks good, thanks
